### PR TITLE
feat(g-canvas): fix webkit ios canvas garbage collection bug

### PR DESCRIPTION
**Pull request by KedAyAyA**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / Document optimization
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Performance improvement
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/antvis/g/issues/726

### 💡 Background and solution

When in a spa application, it's normal to create canvas in a detail page.But when router change, we go to another detail page while creating a new canvas instance.If I create too many, in ios, it will report Total canvas memory use exceeds xxx MB and show me a pure white canvas back.

I find some way to solve this bug, like
- https://stackoverflow.com/questions/52532614/total-canvas-memory-use-exceeds-the-maximum-limit-safari-12
- https://github.com/openlayers/openlayers/issues/9291

And I find the same way to deal in F2 https://github.com/antvis/F2/commit/ea3f84f4bcddb4d296b4977504121dd0a7cbc838.

So I wish G can support this for a better performance in mobile app.

### 📝 Changelog

none

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: packages/g-canvas/src/canvas.ts [hunk 1]
```diff
@@ -228,6 +228,19 @@ class Canvas extends AbstractCanvas {
...
+    const el = this.get('el');
+    if (el) {
+      // 需要清理 canvas 画布内容，否则ios下 创建的canvas垃圾未回收，导致Total canvas memory use exceeds问题
+      // 相关问题列表
+      // https://stackoverflow.com/questions/52532614/total-canvas-memory-use-exceeds-the-maximum-limit-safari-12
+      // https://github.com/openlayers/openlayers/issues/9291
+      el.width = 0;
+      el.height = 0;
+    }
+    el.parentNode.removeChild(el);
```
Comment (hustcc) [inline packages/g-canvas/src/canvas.ts:242]: 这个不需要放到 if 里面嘛？

**Comment by hustcc** [comment]
加一个单测文件吧，参考 bugs 目录。

File: packages/g-canvas/src/canvas.ts (hunk 1 above)
Comment (KedAyAyA) [reply packages/g-canvas/src/canvas.ts:242]: 这个继承的父级上没有放到if 里面，是否这个width跟height的设置也不需要放到if里面呢

Comment (hustcc) [reply packages/g-canvas/src/canvas.ts:242]: 按照这个代码逻辑，确实是要么都放到 if 中，要么就不放到 if 中。那么就都不放到 if 中吧，如果之前就是这么写的话。

Comment (KedAyAyA) [reply packages/g-canvas/src/canvas.ts:240]: ok

**Pull request merged**